In [20]:
import numpy as np
import pandas as pd

In [21]:
dataFile = pd.read_csv('titanic.csv')
type(dataFile)

pandas.core.frame.DataFrame

#### Delete Unnecessary columns

In [22]:
for (columnName,data) in dataFile.iteritems():
    if(columnName not in ["Survived","Pclass","Sex","Age","Fare"]):
        dataFile.drop([columnName],axis='columns',inplace=True)
dataFile

Survived  Pclass     Sex   Age     Fare
0           0       3    male  22.0   7.2500
1           1       1  female  38.0  71.2833
2           1       3  female  26.0   7.9250
3           1       1  female  35.0  53.1000
4           0       3    male  35.0   8.0500
..        ...     ...     ...   ...      ...
886         0       2    male  27.0  13.0000
887         1       1  female  19.0  30.0000
888         0       3  female   NaN  23.4500
889         1       1    male  26.0  30.0000
890         0       3    male  32.0   7.7500

[891 rows x 5 columns]

##### Note: inplace = true will delete cells and return nothing + the deleted cell won't appear in the original datafram. Otherwise, drop() returns dataframe deleting required column

In [36]:
variables  = dataFile.drop('Survived',axis='columns')
targetColumn = dataFile['Survived']

### Now we have separated x and y columns. But we have some columns that contains names rather values. So, we need to map those names to some values or somewhat like that.

In [37]:
variables.Sex = variables.Sex.map({'male':1,'female':2})
variables.head()

Pclass  Sex   Age     Fare
0       3    1  22.0   7.2500
1       1    2  38.0  71.2833
2       3    2  26.0   7.9250
3       1    2  35.0  53.1000
4       3    1  35.0   8.0500

In [38]:
variables.isnull().sum()

Pclass      0
Sex         0
Age       177
Fare        0
dtype: int64

###### Here we can see there are 177 items with no age. So, we need to drop or fill those rows 

In [42]:
variables.Age = variables.Age.fillna(variables.Age.mean())
variables.isnull().sum()

Pclass    0
Sex       0
Age       0
Fare      0
dtype: int64

### Now we go with our prediction function. Until now all kinds of data processing is done

In [ ]:
### Lets split data to training & testing data

In [56]:
from sklearn.model_selection import train_test_split
Xtrain, Xtest, Ytrain, Ytest = train_test_split(variables,targetColumn,test_size=0.2)

In [57]:
from sklearn import tree
model = tree.DecisionTreeClassifier()
model.fit(Xtrain,Ytrain)
PredicTedValues = model.predict(Xtest)
PredicTedValues

array([0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1,
       0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0,
       0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1,
       0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0,
       0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 0, 0, 1, 0,
       0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1,
       1, 1, 1], dtype=int64)

In [58]:
model.score(Xtest,Ytest)

0.8212290502793296